src/System.IO.Compression/src/System/IO/Compression/InflaterZlib.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace System.IO.Compression
+{
+    /// <summary>
+    /// Provides a wrapper around the ZLib decompression API that implements the
+    /// IInflater interface such that this implementation functions may be used interchangeably
+    /// with the Managed implementation found in InflaterManaged.cs
+    /// </summary>
+    internal class InflaterZlib : IInflater
+    {
+        private bool _finished;                             // Whether the end of the stream has been reached
+        private bool _isDisposed;                           // Prevents multiple disposals
+        private ZLibNative.ZLibStreamHandle _zlibStream;    // The handle to the primary underlying zlib stream
+        private GCHandle? _inputBufferHandle;               // The handle to the buffer that provides input to _zlibStream
```

Comment:
This shouldn't need to be nullable.  You should be able to just use ```GCHandle``` and then test whether ```IsAllocated```.